File: GymTrack/Models/FichaTreino.cs
using System.ComponentModel.DataAnnotations;

namespace GymTrack.Models;

public class FichaTreino
{
    public int Id { get; set; }

    public DateTime DataCriacao { get; set; } = DateTime.Now;

    [Required(ErrorMessage = "A ficha deve estar vinculada a um aluno")]
    public int AlunoId { get; set; }
    public Aluno? Aluno { get; set; }

    [Required(ErrorMessage = "A ficha deve possuir um plano de treino")]
    public int PlanoTreinoId { get; set; }
    public PlanoTreino? PlanoTreino { get; set; }

    public ICollection<Exercicio> Exercicios { get; set; } = new List<Exercicio>();
}

File: front/src/models/FichaTreino.ts
import Aluno from "./Aluno";
import Exercicio from "./Exercicio";
import PlanoTreino from "./PlanoTreino";

export default interface FichaTreino {
    id?: number;
    dataCriacao?: string;
    alunoId: number;
    aluno?: Aluno;
    planoTreinoId: number;
    planoTreino?: PlanoTreino;
    exercicios?: Exercicio[];
}

export interface FichaTreinoDto {
    alunoId: number;
    planoTreinoId: number;
    exercicioIds: number[];
}
